streams-contrib/streams-provider-twitter/src/main/java/org/apache/streams/twitter/api/TwitterOAuthRequestInterceptor.java

```diff
@@ -84,16 +85,20 @@ public void process(HttpRequest httpRequest, HttpContext httpContext) throws Htt
     oauthParamMap.put("oauth_version", oauth_version);
 
     String request_host = ((HttpRequestWrapper)httpRequest).getTarget().toString().replace(":443","");
-    String request_path = httpRequest.getRequestLine().getUri().substring(0, httpRequest.getRequestLine().getUri().indexOf('?'));
-    String request_param_line = httpRequest.getRequestLine().getUri().substring(httpRequest.getRequestLine().getUri().indexOf('?')+1);
-    String[] request_params = URLDecoder.decode(request_param_line).split("&");
+    String request_path = httpRequest.getRequestLine().getUri();
 
     Map<String,String> allParamsMap = new HashMap<>(oauthParamMap);
 
-    for( String request_param : request_params ) {
-      String key = request_param.substring(0, request_param.indexOf('='));
-      String value = request_param.substring(request_param.indexOf('=')+1, request_param.length());
-      allParamsMap.put(key, value);
+    if( request_path.contains("?")) {
```

Comment:
having now looked into that class, i agree that using it could reduce number of lines of code.  on the other hand, I just tried to rewrite the class to use it (changed ~100 lines) and the unit tests are failing without a clear remedy. so i'm going to note this on a new ticket and attach a patch file.  it took a LOT of tweaking and testing to get the as-is implementation working...